### PR TITLE
[NETBEANS-5458] Update Batik SVG to version 1.14

### PR DESCRIPTION
**Pull request by sdedic**

### CODE REVIEW

**Comment by eirikbakke** [comment]
Looks fine. I had a look at BATIK-1276, which was the main change you mentioned in the JIRA issue: https://github.com/apache/xmlgraphics-batik/commit/bc6078ca949039e2076cd08b4cb169c84c1179b1

NetBeans' own SVGIcon class does not use the classes in question. But it would be nice and to do something analogous there, i.e.:

```
diff --git a/platform/openide.util.ui.svg/src/org/openide/util/svg/SVGIcon.java b/platform/openide.util.ui.svg/src/org/openide/util/svg/SVGIcon.java
index f6e0347..4bdc7d9 100644
--- a/platform/openide.util.ui.svg/src/org/openide/util/svg/SVGIcon.java
+++ b/platform/openide.util.ui.svg/src/org/openide/util/svg/SVGIcon.java
@@ -38,11 +38,14 @@ import javax.swing.Icon;
 import org.apache.batik.anim.dom.SAXSVGDocumentFactory;
 import org.apache.batik.bridge.BridgeContext;
 import org.apache.batik.bridge.DocumentLoader;
+import org.apache.batik.bridge.ExternalResourceSecurity;
 import org.apache.batik.bridge.GVTBuilder;
+import org.apache.batik.bridge.NoLoadExternalResourceSecurity;
 import org.apache.batik.bridge.UserAgent;
 import org.apache.batik.bridge.UserAgentAdapter;
 import org.apache.batik.ext.awt.image.GraphicsUtil;
 import org.apache.batik.gvt.GraphicsNode;
+import org.apache.batik.util.ParsedURL;
 import org.apache.batik.util.XMLResourceDescriptor;
 import org.openide.util.CachedHiDPIIcon;
 import org.openide.util.Parameters;
@@ -137,7 +140,14 @@ final class SVGIcon extends CachedHiDPIIcon {
             /* Don't provide an URI here; we shouldn't commit to supporting relative links from
             loaded SVG documents. */
             doc = factory.createDocument(null, is);
-            UserAgent userAgent = new UserAgentAdapter();
+            UserAgent userAgent = new UserAgentAdapter() {
+              @Override
+              public ExternalResourceSecurity getExternalResourceSecurity(
+                  ParsedURL resourceURL, ParsedURL docURL)
+              {
+                return new NoLoadExternalResourceSecurity();
+              }
+            };
             DocumentLoader loader = new DocumentLoader(userAgent);
             BridgeContext bctx = new BridgeContext(userAgent, loader);
             try {
```

Since SVGIcon is only currently used on icon files we provide ourselves, this won't currently make a difference. But it would be nice to do if we ever start using SVGIcon to load arbitrary external SVG files. I can add this the next time I'm working on NetBeans.

**Review by eirikbakke — APPROVED** [review]

**Comment by sdedic** [comment]
@eirikbakke Applied ... plus added tests to ensure the xlink:href is really detected.

**Comment by eirikbakke** [comment]
Ah, thank you! Would have done it myself, except building NetBeans takes an hour on my WSL/Windows setup... must get one of those Apple Silicon MacBooks soon.

File: platform/openide.util.ui.svg/test/unit/src/org/openide/util/svg/SVGLoaderImplTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+import org.netbeans.junit.NbTestCase;
+import org.openide.util.ImageUtilities;
+
+/**
+ *
+ * @author sdedic
+ */
+public class SVGLoaderImplTest extends NbTestCase {
```
Comment (eirikbakke) [inline platform/openide.util.ui.svg/test/unit/src/org/openide/util/svg/SVGLoaderImplTest.java:36]: Oh, you did a lot of work to test this--thank you!

**Pull request merged**